devices/Lp3943/Lp3943.cs

```diff
@@ -0,0 +1,185 @@
+			};
+
+			WriteToRegister(register, toWrite);
+		}
+
+		private void WriteToRegister(Register register, byte data)
+		{
+			var message = new byte[2];
+			message[0] = (byte)register;
+			message[1] = data;
+			_i2cDevice.Write(message);
+		}
+
+		private byte FillSelectorRegister(LedState led0, LedState led1, LedState led2, LedState led3)
+		{
+			byte result = 0x00;
+			result |= (byte)(((byte)led3 & 0x03) << 6);
+			result |= (byte)(((byte)led2 & 0x03) << 4);
+			result |= (byte)(((byte)led1 & 0x03) << 2);
+			result |= (byte)((byte)led0 & 0x03);
+			return result;
+		}
+
+		/// <summary>
+		/// Sets the dimming on a register
+		/// </summary>
+		/// <param name="register">register to dim</param>
+		/// <param name="frequency">frequency in Hz</param>
+		/// <param name="dimPercentage">percentage</param>
+		public void DimRegister(DimRegister register, int frequency, int dimPercentage)
+		{
+			if (frequency is > 160 or < 1)
+				throw new ArgumentOutOfRangeException(nameof(frequency), "The frequency has to be between 1 and 160 Hz.");
+
+			if (dimPercentage is > 100 or < 0)
+				throw new ArgumentOutOfRangeException(nameof(dimPercentage));
+
+			SetFrequency(register, frequency);
+			SetDimPercentage(register, dimPercentage);
+		}
+
+		private void Update()
+		{
+			var ls0 = FillSelectorRegister(_ledStates[0], _ledStates[1], _ledStates[2], _ledStates[3]);
+			var ls1 = FillSelectorRegister(_ledStates[4], _ledStates[5], _ledStates[6], _ledStates[7]);
+			var ls2 = FillSelectorRegister(_ledStates[8], _ledStates[9], _ledStates[10], _ledStates[11]);
+			var ls3 = FillSelectorRegister(_ledStates[12], _ledStates[13], _ledStates[14], _ledStates[15]);
+
+			WriteToRegister(Register.Ls0, ls0);
+			WriteToRegister(Register.Ls1, ls1);
+			WriteToRegister(Register.Ls2, ls2);
+			WriteToRegister(Register.Ls3, ls3);
+		}
+
+		/// <summary>
+		/// Resets the Lp3943 chip
+		/// </summary>
+		public void Reset()
+		{
+			_controller.Write(_pinReset, PinValue.Low);
+			Thread.Sleep(5);
+			_controller.Write(_pinReset, PinValue.High);
+		}
+
+		/// <summary>
+		/// Sets led to assigned mode
+		/// </summary>
+		/// <param name="led">led to assign</param>
+		/// <param name="ledState">state to give the led</param>
+		public void SetLed(int led, LedState ledState)
+		{
+			if (led is > 16 or < 0)
+				throw new ArgumentOutOfRangeException(nameof(led), "Led can only be a value between 0 and 16");
+			_ledStates[led] = ledState;
+			Update();
+		}
+
+		/// <summary>
+		/// Sets the LEDs in the array to the assigned mode
+		/// </summary>
+		/// <param name="leds">LEDs to assign</param>
+		/// <param name="ledState">state to give the LEDs</param>
+		public void SetLed(int[] leds, LedState ledState)
+		{
+			if (leds is null)
+				throw new ArgumentNullException(nameof(leds));
+
+			foreach (var led in leds)
+			{
+				if (led is <= 15 and >= 0)
+				{
+					_ledStates[led] = ledState;
+				}
```

Comment:
else throws an exception?

Reply:
Leaving a warning would be best but an exception is fine as well I guess

Reply:
thanks. And to gain space, you can remove the text from the exception.